base/oauth.go

```diff
@@ -160,6 +165,11 @@ func GetOAuthClient(
 
 	// we need to request new authorization
 	if token == nil {
+		// skip authentication if specified
+		if opts.SkipAuthentication {
```

Comment:
It will be used in the non-auth flows such as `!gcal list-calendars work`. If `client == nil`, it will give you an error message instead of trying to authenticate.

Reply:
Hmm it's a little weird that `GetOAuthClient` can be nooped, the caller just shouldn't call the function if it doesnt need the client

Reply:
I see what you mean

Reply:
This kind of complicates things, as any command can now trigger an authentication. This means I need to refactor some of the callback stuff so that it can take an identifier.